tap_pendo/streams.py

```diff
@@ -416,22 +416,28 @@ def send_request_get_results(self, req):
         # If value is 0,"0", "" or None then it will set default to default to 300.0 seconds if not passed in config.
         config_request_timeout = self.config.get('request_timeout')
         request_timeout = config_request_timeout and float(config_request_timeout) or REQUEST_TIMEOUT # pylint: disable=consider-using-ternary
-        with session.send(req, stream=True, timeout=request_timeout) as resp:
```

Comment:
Backoff scenario was not working for yield statement when written under  `with` operator hence we updated the logic to handle backoff scenario properly.

Reply:
Added resp.closed() as per suggestion.